Criterion/Legacy/Search/Gateway/CriterionHandler/PublishedOnDayOfWeek.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\CookbookBundle\Criterion\Legacy\Search\Gateway\CriterionHandler;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter;
+use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriterionHandler;
+use EzSystems\CookbookBundle\Criterion as CustomCriterion;
+
+class PublishedOnDayOfWeek extends CriterionHandler
+{
+    const WEEKDAY_MAP = [
+        CustomCriterion\PublishedOnDayOfWeek::SUN => 1,
+        CustomCriterion\PublishedOnDayOfWeek::MON => 2,
+        CustomCriterion\PublishedOnDayOfWeek::TUE => 3,
+        CustomCriterion\PublishedOnDayOfWeek::WED => 4,
+        CustomCriterion\PublishedOnDayOfWeek::THU => 5,
+        CustomCriterion\PublishedOnDayOfWeek::FRI => 6,
+        CustomCriterion\PublishedOnDayOfWeek::SAT => 7,
+    ];
+    /**
+     * Check if this criterion handler accepts to handle the given criterion.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     *
+     * @return bool
+     */
+    public function accept(Criterion $criterion)
+    {
+        return $criterion instanceof CustomCriterion\PublishedOnDayOfWeek;
+    }
+
+    /**
+     * @param \eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter $converter
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     * @param array $languageSettings
+     * @return string
+     */
+    public function handle(
+        CriteriaConverter $converter,
+        SelectQuery $query,
+        Criterion $criterion,
+        array $languageSettings
+    ) {
+        $column = $this->dbHandler->quoteColumn('published', 'ezcontentobject');
+        $value = array_map([$this, 'mapCriterionValue'], $criterion->value);
+
+        return $query->expr->in(
+            // Note: MySQL-specific
+            "DAYOFWEEK(FROM_UNIXTIME($column))",
```

Comment:
This is very DBMS-specific (along with `mapCriterionValue` because numbering of weekdays varies between PHP and DMBS-es).
Unfortunately I'm not sure how to make it more generic here. We could document that Legacy criterion handler depends on DBMS and should be created separately for each DBMS and a proper one should be chosen, but maybe there's a way to use here different strategies for different DBMS-es?

ping @andrerom